docs/200_ContextGuard.md

```diff
@@ -0,0 +1,40 @@
+# Context Guard
+
+The context guard add an additional layer of protection and controll ontop of the set of your indexing rules. It allows you to filter the result and avoid indexing unvanted content.
+
+To  implement a guard add a service that would implement `ContextGuardInterface`.
+
+## Example Setup
+
+``` yaml
+services:
+  AppBundle\DynamicSearch\Guard\DefaultContextGuard:
+    tags:
+      - { name: dynamic_search.context_guard }
+```
+This simple guard will prevent the document with the id 1 (it is the root document in the pimcore document tree) to be added to the index.
+
+``` php
+<?php
+
+namespace AppBundle\DynamicSearch\Guard;
```

Comment:
Please use `App` Namespace here (Symfony 5 Standard)

Reply:
Thx, I will check al lthe steps again on a fresh setup and add some updates